### DISCUSSION TITLE
values returned from resolver do not update form values

### DISCUSSION THREAD

**Discussion by kpodmasko** [react-hook-form/react-hook-form#9353] [Q&A] [answered]
### Version Number

7.39.1

### Codesandbox/Expo snack

https://codesandbox.io/s/custom-hook-with-resolver-v7-forked-ulv288?file=/src/App.js

### Steps to reproduce

1. open codesandbox

### Expected behaviour

I want to validate and fix defaultValues (if they come from api)
I expect that trigger in useEffect will run validation. It will run resolver. Resolver will validate defaultValues. If some fields are invalid they will be tried to fix (use default value from passed schema).

EXPECTATION: After fixing, resolver will return new values that will be applied to form
BUG: After fixing, resolver returns new values and they are not applied to form

console.log shows that values before fix and after fix are different, but they are no applied to form.
Also I tried to do ugly form rerender with timeout after resolver running, but it does not help to update form values

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
> After fixing, resolver will return new values that will be applied to form

I am a bit confused with your `setState` after `trigger()`, if you want to update the form's default values consider using `reset` instead.

**Reply by kpodmasko** [reply]
ugly `setState` is only for example :)

I missed that we can provide values to `reset`. thx for help

**Comment by riotrah** [comment]
I think the confusion in this discussion illustrates something I'm confused about as well, which isn't described in the docs:

What _is_ the purpose of the `values` key in a resolver's return object? If form's state is not replaced with `values`, then what is done with `values`?
